### PR TITLE
Postpone: Fixes when method is hardlink but it should move the media

### PR DESCRIPTION
**Pull request by fernandog**
Improve a lot of duplicate code and:

1) Commit 1 tested ok ~~needs more testing~~: Fixes when method is hardlink but it should move the media (It already works for not postpone)

2) Commit 2 tested ok: Fixes when medusa download a subtitle and when I remove torrent with data, folder is not deleted because of subtitles (that wasn't part of the torrent)

### CODE REVIEW

File: medusa/post_processor.py [hunk 1]
```diff
@@ -443,20 +434,7 @@ def _int_copy(cur_file_path, new_file_path):
...
-        Hardlink file and set proper permissions.
-
-        :param file_path: The full path of the media file to move
-        :param new_path: Destination path where we want to create a hard linked file
-        :param new_base_name: The base filename (no extension) to use during the link. Use None to keep the same name.
-        :param associated_files: Boolean, whether we should move similarly-named files too
-        """
-        def _int_hard_link(cur_file_path, new_file_path):
-
+        def hardlink(cur_file_path, new_file_path):
```
Comment (fernandog) [inline medusa/post_processor.py]: its a callback function. shut up

File: medusa/post_processor.py [hunk 2]
```diff
@@ -466,20 +444,7 @@ def _int_hard_link(cur_file_path, new_file_path):
...
-        Move file, symlink source location back to destination, and set proper permissions.
-
-        :param file_path: The full path of the media file to move
-        :param new_path: Destination path where we want to move the file to create a symbolic link to
-        :param new_base_name: The base filename (no extension) to use during the link. Use None to keep the same name.
-        :param associated_files: Boolean, whether we should move similarly-named files too
-        """
-        def _int_move_and_sym_link(cur_file_path, new_file_path):
-
+        def symlink(cur_file_path, new_file_path):
```
Comment (fernandog) [inline medusa/post_processor.py]: its a callback function. shut up

File: medusa/post_processor.py [hunk 3]
```diff
@@ -420,20 +424,7 @@ def _int_move(cur_file_path, new_file_path):
...
-        Copy file and set proper permissions.
-
-        :param file_path: The full path of the media file to copy
-        :param new_path: Destination path where we want to copy the file to
-        :param new_base_name: The base filename (no extension) to use during the copy. Use None to keep the same name.
-        :param associated_files: Boolean, whether we should copy similarly-named files too
-        """
-        def _int_copy(cur_file_path, new_file_path):
-
+        def copy(cur_file_path, new_file_path):
```
Comment (fernandog) [inline medusa/post_processor.py]: its a callback function. shut up

File: medusa/post_processor.py [hunk 4]
```diff
@@ -490,8 +455,9 @@ def _int_move_and_sym_link(cur_file_path, new_file_path):
                           (cur_file_path, new_file_path, e), logger.ERROR)
                 raise
 
+        subtitle_action = move if self.process_method in ['hardlink', 'symlink'] else eval(self.process_method)
         self._combined_file_operation(file_path, new_path, new_base_name, associated_files,
-                                      action=_int_move_and_sym_link, subtitles=subtitles)
+                                      action=eval(action), subtitle_action=subtitle_action, subtitles=subtitles)
```
Comment (p0psicles) [inline medusa/post_processor.py]: I need to read your code on a desktop. But I suspect your using eval here, while that's not needed. You can also map the different methods in a dict.

Comment (fernandog) [inline medusa/post_processor.py]: I needed otherwise i will get unicode not callable

**Comment by fernandog** [comment]
i already PPed more than 10 files with this PR (.rar torrents and normal torrents)

the only think i don't know how to fix is this codacy warnings. any idea?

File: medusa/post_processor.py [hunk 5]
```diff
@@ -455,7 +455,9 @@ def symlink(cur_file_path, new_file_path):
                           (cur_file_path, new_file_path, e), logger.ERROR)
                 raise
 
-        subtitle_action = move if self.process_method in ['hardlink', 'symlink'] else eval(self.process_method)
+        action = {'copy': copy, 'move': move, 'hardlink': hardlink, 'symlink': symlink}.get(self.process_method)
+        # Subtitle action should be move in case of hardlink|symlink as downloaded subtitle is not part of torrent
+        subtitle_action = {'copy': copy, 'move': move, 'hardlink': move, 'symlink': move}.get(self.process_method)
```
Comment (fernandog) [inline medusa/post_processor.py]: @p0psicles not using `eval()` and fixed codacy issue about function not used =)

File: medusa/post_processor.py [hunk 6]
```diff
@@ -455,9 +455,11 @@ def symlink(cur_file_path, new_file_path):
...
         self._combined_file_operation(file_path, new_path, new_base_name, associated_files,
-                                      action=eval(action), subtitle_action=subtitle_action, subtitles=subtitles)
+                                      action=action, subtitle_action=subtitle_action, subtitles=subtitles)
```
Comment (fernandog) [inline medusa/post_processor.py]: @p0psicles not using `eval()` and fixed codacy issue about function not used =)

**Comment by fernandog** [comment]
if no one else have comments on this i will merge in about 8 hours

**Comment by fernandog** [comment]
All good:

RARed Torrents. Moved media and moved subtitle
```
2017-02-08 23:26:24 DEBUG    FINDSUBTITLES :: [b0fea9f] Moving file from /media/SAMSUNG/media/downloaded/series/DCs.Legends.of.Tomorrow.S02E11.1080p.HDTV.X264-DIMENSION/DCs.Legends.of.Tomorrow.S02E11.1080p.HDTV.X264-DIMENSION.pt-BR.srt to /media/SAMSUNG/media/series/Legends of Tomorrow/Season 2/DCs.Legends.of.Tomorrow.S02E11.1080p.HDTV.X264-DIMENSION.pt-BR.srt 
2017-02-08 23:26:24 DEBUG    FINDSUBTITLES :: [b0fea9f] Moving file from /media/SAMSUNG/media/downloaded/series/DCs.Legends.of.Tomorrow.S02E11.1080p.HDTV.X264-DIMENSION/DCs.Legends.of.Tomorrow.S02E11.1080p.HDTV.X264-DIMENSION.mkv to /media/SAMSUNG/media/series/Legends of Tomorrow/Season 2/DCs.Legends.of.Tomorrow.S02E11.1080p.HDTV.X264-DIMENSION.mkv 
2017-02-08 22:37:20 DEBUG    FINDSUBTITLES :: [b0fea9f] Moving file from /media/SAMSUNG/media/downloaded/series/Quantico.S02E11.720p.HDTV.X264-DIMENSION/Quantico.S02E11.720p.HDTV.X264-DIMENSION.pt-BR.srt to /media/SAMSUNG/media/series/Quantico/Season 2/Quantico.S02E11.720p.HDTV.X264-DIMENSION.pt-BR.srt 
2017-02-08 22:37:20 DEBUG    FINDSUBTITLES :: [b0fea9f] Moving file from /media/SAMSUNG/media/downloaded/series/Quantico.S02E11.720p.HDTV.X264-DIMENSION/Quantico.S02E11.720p.HDTV.X264-DIMENSION.mkv to /media/SAMSUNG/media/series/Quantico/Season 2/Quantico.S02E11.720p.HDTV.X264-DIMENSION.mkv 
```

Non-RARed torrents: Hard linked media and moved subtitle:
```
2017-02-08 19:37:24 DEBUG    FINDSUBTITLES :: [b0fea9f] Hard linking file from /media/SAMSUNG/media/downloaded/series/Marvels.Agents.of.S.H.I.E.L.D.S04E13.1080p.WEB-DL.DD5.1.H264-RARBG/Marvels.Agents.of.S.H.I.E.L.D.S04E13.1080p.WEB-DL.DD5.1.H264-RARBG.mkv to /media/SAMSUNG/media/series/Marvel's Agents of S.H.I.E.L.D/Season 4/Marvels.Agents.of.S.H.I.E.L.D.S04E13.1080p.WEB-DL.DD5.1.H264-RARBG.mkv
2017-02-08 19:37:24 DEBUG    FINDSUBTITLES :: [b0fea9f] Moving file from /media/SAMSUNG/media/downloaded/series/Marvels.Agents.of.S.H.I.E.L.D.S04E13.1080p.WEB-DL.DD5.1.H264-RARBG/Marvels.Agents.of.S.H.I.E.L.D.S04E13.1080p.WEB-DL.DD5.1.H264-RARBG.pt-BR.srt to /media/SAMSUNG/media/series/Marvel's Agents of S.H.I.E.L.D/Season 4/Marvels.Agents.of.S.H.I.E.L.D.S04E13.1080p.WEB-DL.DD5.1.H264-RARBG.pt-BR.srt

2017-02-08 16:08:57 DEBUG    FINDSUBTITLES :: [b0fea9f] Hard linking file from /media/SAMSUNG/media/downloaded/series/The.Flash.2014.S03E12.1080p.HDTV.X264-DIMENSION/The.Flash.2014.S03E12.1080p.HDTV.X264-DIMENSION.mkv to /media/SAMSUNG/media/series/The Flash (2014)/Season 3/The.Flash.2014.S03E12.1080p.HDTV.X264-DIMENSION.mkv
2017-02-08 16:08:57 DEBUG    FINDSUBTITLES :: [b0fea9f] Moving file from /media/SAMSUNG/media/downloaded/series/The.Flash.2014.S03E12.1080p.HDTV.X264-DIMENSION/The.Flash.2014.S03E12.1080p.HDTV.X264-DIMENSION.pt-BR.srt to /media/SAMSUNG/media/series/The Flash (2014)/Season 3/The.Flash.2014.S03E12.1080p.HDTV.X264-DIMENSION.pt-BR.srt

```

**Pull request merged**